flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultPartitionFactory.java

```diff
@@ -240,6 +242,8 @@ public ResultPartition create(
                                                             .SpillingStrategyType.SELECTIVE)
                                     .build(),
                             bufferCompressor,
+                            // Only hybrid full result partition support broadcast optimization.
+                            isBroadcast && type == ResultPartitionType.HYBRID_FULL,
```

Comment:
Instead of disabling this optimization for selective spilling, I'd suggest to always use full spilling when the result partition is broadcast.